integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/SparkApplicationExecutionContext.java

```diff
@@ -39,6 +45,11 @@ class SparkApplicationExecutionContext implements ExecutionContext {
   private final EventEmitter eventEmitter;
   private final OpenLineageRunEventBuilder runEventBuilder;
   private final OpenLineage openLineage = new OpenLineage(Versions.OPEN_LINEAGE_PRODUCER_URI);
+  private final List<JobNameProvider> jobNameProviders =
```

Comment:
I think if we're hard forcing the precedence order, we need to comment it here.

Reply:
I've moved this part to a separate class - `ApplicationJobNameResolver` where we have hardcoding part explained and the logic.